step/archive.go

```diff
@@ -48,6 +53,46 @@ func runArchiveCommand(archiveCmd *xcodebuild.CommandBuilder, useXcpretty bool,
 	progress.SimpleProgress(".", time.Minute, func() {
 		err = archiveRootCmd.Run()
 	})
+	out := output.String()
+
+	return output.String(), wrapXcodebuildCommandError(archiveCmd, out, err)
+}
+
+type Printable interface {
+	PrintableCmd() string
+}
+
+func wrapXcodebuildCommandError(cmd Printable, out string, err error) error {
+	if err == nil {
+		return nil
+	}
+
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) {
+		reasons := findErrors(out)
+		if len(reasons) > 0 {
+			return fmt.Errorf("command (%s) failed with exit status %d: %w", cmd.PrintableCmd(), exitErr.ExitCode(), errors.New(strings.Join(reasons, "\n")))
+		}
+		return fmt.Errorf("command (%s) failed with exit status %d", cmd.PrintableCmd(), exitErr.ExitCode())
+	}
+
+	return fmt.Errorf("executing command (%s) failed: %w", cmd.PrintableCmd(), err)
+}
+
+func findErrors(out string) []string {
+	var errors []string
+
+	scanner := bufio.NewScanner(strings.NewReader(out))
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "error: ") {
+			errors = append(errors, line)
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return nil
+	}
 
-	return output.String(), err
+	return errors
```

Comment:
This is only a nitpick comment so do not pay too much attention to it. These lines are shared between the archive and ipa export commands. In such cases I like to extract the common part into a separate file to also visually separate the code pieces. I know this is just a style and Go is not fussy about it but what if we move these into the already existing `step/errors.go` file?

Reply:
Moved to the `errors.go` file.